src/AbstractServer.php

```diff
@@ -52,7 +52,7 @@ protected function startServer($minPort, $maxPort, &$port)
      */
     protected function processEventData($data)
     {
-        $dataObject = json_decode($data, true);
+        $dataObject = json_decode((string) $data, true);
```

Comment:
Puvodne jsem chtel navrhnout pridat `string` jako typ k argumentu metody. Ale kdyz koukam, jak se pouziva, byla by to zase takova rabbit hole (aby to melo smysl), tak mozna bych spis rekl, udelat tu `if(!is_string($data)) throw InvalidMessageException(...)`?

Reply:
No to řeší ten následující řádek, pokud to nebude string, tak to nebude ani validní JSON a zpracuje se to v té výjimice

Reply:
No s tím processEventData(string $data) mě to taky samozřejmě napadlo, ale dlouho mě to nadšení nevydrželo :)

Reply:
No kdyby ti tam prisel objekt (bez `__toString()`) nebo pole, tak ti to pretypovani na string padne na nejaky type conversion vyjimce. Ale to tu asi nehrozi, jinak by to padalo uz ted.

Znas me, ja bych tam nabusil typy odzhora az dolu :slightly_smiling_face:

Reply:
Jasně no, to by se nemělo stát jelikož to tam leze z nějakýho get stream contents. 

Typy odzhora dolů bych taky rád, ale popořadě. Tohleto je dost letitý kousek :) .